ReactWindows/ReactNative/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -20,5 +45,281 @@ public override string Name
+                    break;
+                case JTokenType.Integer:
+                    _dataContainer.Values[key] = token.Value<long>();
+                    break;
+                case JTokenType.Float:
+                    _dataContainer.Values[key] = token.Value<double>();
+                    break;
+                case JTokenType.String:
+                    _dataContainer.Values[key] = token.Value<string>();
+                    addTokenTypeToContainer(key, _DataType.String);
+                    break;
+                case JTokenType.Array:
+                    _dataContainer.Values[key] = token.ToString();
+                    addTokenTypeToContainer(key, _DataType.Array);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Gets related value for a specific key from ApplicationDataContainer as JToken.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        private JToken getTokenFromContainer(string key)
+        {
+            var value = _dataContainer.Values[key];
+            if (value == null)
+            {
+                return JValue.CreateNull();
+            }
+            else
+            {
+                var t = value.GetType();
+                if (t == typeof(bool) || t == typeof(long) || t == typeof(double))
+                {
+                    return JToken.FromObject(value);
+                }
+                else if (t == typeof(string))
+                {
+                    var type = getTokenTypeFromContainer(key);
+                    if (type == _DataType.Array)
+                    {
+                        return JToken.Parse((string)value);
+                    }
+                    else if (type == _DataType.String)
+                    {
+                        return JToken.FromObject((string)value);
+                    }
+                    else
+                    {
+                        return JValue.CreateNull();
+                    }
+                }
+                else
+                {
+                    return JValue.CreateNull();
+                }
+            }
+        }
+
+        // <summary>
+        /// Adds token's type to ApplicationDataContainer with the specified key.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        /// <param name="type">The type.</param>
+        private void addTokenTypeToContainer(string key, _DataType type)
+        {
+            _typeContainer.Values[key] = (int)type;
+        }
+
+        /// <summary>
+        /// Gets token's type for a specific key from ApplicationDataContainer.
+        /// </summary>
+        /// <param name="key">The key.</param>
+        private _DataType getTokenTypeFromContainer(string key)
+        {
+            var value = _typeContainer.Values[key];
+            if (value == null)
+            {
```

Comment:
Shouldn't you use TryGetValue instead in case someone adds special behavior for a key mapped to a null value?